src/click/termui.py

```diff
@@ -50,10 +46,32 @@
 _ansi_reset_all = "\033[0m"
 
 
-def hidden_prompt_func(prompt: str) -> str:
+def visible_prompt_func(prompt: str, err: bool = False) -> str:
+    """
+    .. versionadded:: 8.0.2
+        Added the `err` parameter.
+
+    """
+    try:
+        sys.stdout = sys.__stderr__ if err else sys.__stdout__
```

Comment:
This is not correct. Click purposefully uses `sys.stdout`, not the original `__stdout__`, in order to allow overriding `stdout`, such as in tests.

Reply:
I am not sure I understand. The purpose of `sys.__stdout__` et al is to be the [original value of `sys.stdout`, and the Python docs recommend to use it to reset them to that original value](https://docs.python.org/3/library/sys.html#sys.__stdin__).

Reply:
Tools like pytest and Click's CliRunner already override `sys.stdout`. We would want to use those, not the originals.

Reply:
OK, understood.